### ISSUE TITLE
HTTP Client throwing read timeouts when using HTTP connection pool

### ISSUE THREAD

**Issue by aseychell** [micronaut-projects/micronaut-core#9933]
### Expected Behavior

When using the HTTP client and the server is responsive, we should not be getting read timeouts

### Actual Behaviour

A read timeout is thrown because the underlying connection pool did not make progress with the pending requests.

```
2023-09-29T09:04:48.419Z [boundedElastic-1] ERROR reactor.core.publisher.Operators - Operator called default onErrorDropped
reactor.core.Exceptions$ErrorCallbackNotImplemented: io.micronaut.http.client.exceptions.ReadTimeoutException: Read Timeout
Caused by: io.micronaut.http.client.exceptions.ReadTimeoutException: Read Timeout
	at io.micronaut.http.client.exceptions.ReadTimeoutException.<clinit>(ReadTimeoutException.java:26)
	at io.micronaut.http.client.netty.DefaultHttpClient.lambda$exchangeImpl$28(DefaultHttpClient.java:1135)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:94)
	at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124)
	at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.handleTimeout(FluxTimeout.java:295)
	at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.doTimeout(FluxTimeout.java:280)
	at reactor.core.publisher.FluxTimeout$TimeoutTimeoutSubscriber.onNext(FluxTimeout.java:419)
	at reactor.core.publisher.FluxOnErrorReturn$ReturnSubscriber.onNext(FluxOnErrorReturn.java:162)
	at reactor.core.publisher.MonoDelay$MonoDelayRunnable.propagateDelay(MonoDelay.java:271)
	at reactor.core.publisher.MonoDelay$MonoDelayRunnable.run(MonoDelay.java:286)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Steps To Reproduce

1. Within a @Scheduled method, trigger an HTTP call to an external API
2. Configure the Netty HTTP client to have the connection pool enabled
3. At some point, it will start giving read-timeouts

### Example Application

https://github.com/aseychell/micronaut4-http-readtimeouts

### Version

4.1.2

**Comment by rorueda** [comment]
So, I spent some time looking at this and I think the problem happens because the pool implementation is not handling situations where the channel becomes unusable:
- The channel becomes inactive but apparently the `channelInactive` handler is not called. This happened when running the sample application, but I couldn't reproduce in a unit test.
- A `GOAWAY` is received but the channel does not become inactive - this depends on the error code of the `GOAWAY`. I could reproduce this in a unit test by finishing the server channel. This results in a different exception, but the fact is that the `ConnectionManager` is trying to open new streams after receiving a `GOAWAY`.

When I have more spare time I might try to improve the tests or attempt a fix, but thought the extra info could be useful.

**Comment by yawkat** [comment]
@rorueda #10007 should stop use of the connection after goaway at least. that might resolve this issue.

**Comment by rorueda** [comment]
@yawkat Great.

I dug a bit more and was able to understand and reproduce the first situation in an unit test. If you think it is worth I can open a PR (see https://github.com/rorueda/micronaut-core/commit/3c506b7d4d3da3f71fe59856d705ffb3b12765ff).

**Comment by yawkat** [comment]
thanks for tracking that down @rorueda, good catch. please make a pr so you can get proper credit, though i want to change the fix slightly.

**Comment by rorueda** [comment]
No problem, @yawkat. PR opened.

**Comment by yawkat** [comment]
going to close this now that the two fixes are merged. i hope that's all of the bugs in this issue.

**Comment by stefanos-kalantzis** [comment]
Do we know when this is going to be released?

I'm assuming that 4.1.6 refers to the micronaut-platform, and not core which is already at 4.1.10, right?

**Comment by yawkat** [comment]
it will be fixed in core 4.1.11 and platform 4.1.6
